addons/thtracker/thtracker.lua

```diff
@@ -61,33 +62,51 @@ windower.register_event('addon command', function(command, ...)
         if posx and posy then
             th:pos(posx, posy)
         end
-
     elseif command == "hide" then
         th:hide()
-
     elseif command == 'show' then
         th:show()
-
     else
         print('th help : Shows help message')
         print('th pos <x> <y> : Positions the list')
         print('th hide : Hides the box')
         print('th show : Shows the box')
-
     end
 end)
 
 windower.register_event('incoming text', function(original, new, color)
-    local name, count = original:match('Additional effect: Treasure Hunter effectiveness against the (.*) increases to (%d+).')
-    if name and count then
-        mob = name
-        th:text(' '..name..'\n TH: '..count);
-        th:show()
-    end
-
-    name = original:match('%w+ defeats the (.*).')
-    if name and mob == name then
-        th:text('')
-        th:hide()
-    end
+	original = original:strip_format()
+	if string.find(original, "Treasure Hunter") then
+		name, count = original:match('Additional effect: Treasure Hunter effectiveness against the (.*) increases to (%d+).')
+		if not name and not count then
+			name, count = original:match('Additional effect: Treasure Hunter effectiveness against (.*) increases to (%d+).')
+		end
```

Comment:
The windower.regex was exactly what I was looking for (obviously didn't look hard enough). That will be so much better as it should be similar, if not the same, to what I use for JavaScript. Thanks.